cpp/ScanChain_Zp.C

```diff
@@ -649,6 +675,36 @@ int ScanChain(TChain *ch, double genEventSumw, TString year, TString process) {
       double puppimet_pt  = puppimet.first;
       double puppimet_phi = puppimet.second;
 
+      // Account for tunepRelPt if useTuneP
+      vector<LorentzVector> Muon_p4 = {};
+      vector<double> Muon_pt = {};
+      vector<double> Muon_tkRelIso = {};
+      TVector2 pfmet_temp;
+      pfmet_temp.SetMagPhi(pfmet_pt,pfmet_phi);
+      for ( unsigned int im=0; im < nt.nMuon(); im++ ) {
+	Muon_p4.push_back(LorentzVector(nt.Muon_pt().at(im),nt.Muon_eta().at(im),nt.Muon_phi().at(im),nt.Muon_mass().at(im)));
+	Muon_pt.push_back(nt.Muon_pt().at(im));
+	Muon_tkRelIso.push_back(nt.Muon_tkRelIso().at(im));
+	if ( nt.Muon_highPtId().at(im)>=2 ) {
+	  if ( useTuneP ) {
+	    TVector2 muon_temp;
+	    TVector2 muon_temp_tunep;
+	    muon_temp.SetMagPhi(Muon_pt[im],nt.Muon_phi().at(im));
+	    Muon_p4[im] = LorentzVector(Muon_pt[im],nt.Muon_eta().at(im),nt.Muon_phi().at(im),nt.Muon_mass().at(im));
+	    Muon_pt[im] *= nt.Muon_tunepRelPt().at(im);
+	    muon_temp_tunep.SetMagPhi(Muon_pt[im],nt.Muon_phi().at(im));
+	    // Correct PF MET for highPtId muon candidates
+	    pfmet_temp+=muon_temp;
+	    pfmet_temp-=muon_temp_tunep;
+	  }
+	  else {
+	    Muon_tkRelIso[im] *= nt.Muon_tunepRelPt().at(im);
+	  }
+	}
+      }
+      pfmet_pt  = pfmet_temp.Mod();
+      pfmet_phi = pfmet_temp.Phi();
```

Comment:
Looking at the [plots](https://uaf-10.t2.ucsd.edu/~mmasciov/Zprime/plots_May-03-2022/pfmet_phi_sel8_mllinclusive_nBTag1_mc+data_logY.png), do we understand why this comes out as > 0? I see there is no `fabs` applied and I would have expected that the standard convention would be `-π < φ < π`.

Reply:
Understood as coming from TVector2::Phi(). Fixed.

Reply:
This is an example where I only rerun on Y3(200):
http://uaf-10.t2.ucsd.edu/~mmasciov/Zprime/plots_May-04-2022/pfmet_phi_sel8_mllinclusive_nBTag1p_mc+data_logY.png
For Y3(200), now MET_phi looks as expected.